src/pipeline/datatype/SpatialImgDetections.cpp

```diff
@@ -1,11 +1,168 @@
+    this->boundingBox = boundingBox;
+
+    std::array<float, 4> outerPoints = this->boundingBox->getOuterRect();
+    xmin = outerPoints[0];
+    ymin = outerPoints[1];
+    xmax = outerPoints[2];
+    ymax = outerPoints[3];
+}
+
+RotatedRect SpatialImgDetection::getBoundingBox() const {
+    if(boundingBox.has_value()) {
+        return boundingBox.value();
+    } else if(xmin == 0.f && xmax == 0.f && ymin == 0.f && ymax == 0.f) {
+        throw std::runtime_error("All bounding box values are zero, no bounding box can be built.");
+    }
+
+    // construct from legacy values
+    RotatedRect rect = RotatedRect{dai::Rect{Point2f{xmin, ymin}, Point2f{xmax, ymax}}};
+    return rect;
+}
+
+void SpatialImgDetection::setOuterBoundingBox(const float xmin, const float ymin, const float xmax, const float ymax) {
+    this->xmin = xmin;
+    this->ymin = ymin;
+    this->xmax = xmax;
+    this->ymax = ymax;
+
+    Point2f topLeft{xmin, ymin};
+    Point2f bottomRight{xmax, ymax};
+
+    this->boundingBox = RotatedRect{Rect{topLeft, bottomRight}};
+}
+
+void SpatialImgDetection::setKeypoints(const SpatialKeypointsList kp) {
+    this->keypoints = kp;
+}
+
+void SpatialImgDetection::setKeypoints(const std::vector<SpatialKeypoint> kp) {
+    this->keypoints = SpatialKeypointsList(kp);
+}
+
+void SpatialImgDetection::setKeypoints(const std::vector<SpatialKeypoint> kps, const std::vector<std::array<uint32_t, 2>> edges) {
+    this->keypoints = SpatialKeypointsList(kps, edges);
+}
+
+void SpatialImgDetection::setKeypoints(const std::vector<Point3f> kps3) {
+    this->keypoints = SpatialKeypointsList();
+    this->keypoints->setKeypoints(kps3);
+}
+
+void SpatialImgDetection::setSpatialCoordinate(Point3f spatialCoordinates) {
+    this->spatialCoordinates = spatialCoordinates;
+}
+
+dai::ImgDetection SpatialImgDetection::getImgDetection() const {
+    dai::ImgDetection imgDetection(getBoundingBox(), labelName, confidence, label);
+
+    if(keypoints.has_value()) {
+        dai::KeypointsList converted;
+        std::vector<dai::Keypoint> convertedKeypoints;
+        convertedKeypoints.reserve(keypoints->size());
+        for(const auto& spatialKeypoint : keypoints->getKeypoints()) {
+            convertedKeypoints.emplace_back(spatialKeypoint.imageCoordinates, spatialKeypoint.confidence, spatialKeypoint.label);
+        }
+        converted.setKeypoints(std::move(convertedKeypoints), keypoints->getEdges());
+        imgDetection.keypoints = std::move(converted);
+    }
+
+    return imgDetection;
+}
+
+std::vector<SpatialKeypoint> SpatialImgDetection::getKeypoints() const {
+    if(keypoints.has_value()) {
+        return keypoints->getKeypoints();
+    } else {
+        return {};
+    }
+}
+
+std::vector<dai::Point3f> SpatialImgDetection::getKeypointSpatialCoordinates() const {
+    if(!this->keypoints.has_value()) {
+        return {};
+    }
+    return this->keypoints->getSpatialCoordinates();
+}
```

Comment:
To follow the  same style as the method above and bellow:

```suggestion
std::vector<dai::Point3f> SpatialImgDetection::getKeypointSpatialCoordinates() const {
    if(keypoints.has_value()) {
        return keypoints->getSpatialCoordinates();;
    } else {
        return {};
    }
}
```